File: Java/Platform/Basics/FileChannelTest/Transformer.java
class Transformer {

	public static void transform(java.nio.ByteBuffer data, int count) {
		for(int i = 0, j = count - 1; i < j; ++i, --j){
			byte ib = data.get(i);
			byte jb = data.get(j);
			data.put(i, jb);
			data.put(j, ib);
		}
	}
}

File: dotnet/Platform/Basics/FileIOTest/DemoApp/Transformer.cs
static class Transformer
{
    public static void Encode(Span<byte> bytes, int count)
    {
        for(int i = 0; i < count; ++i)
            bytes[i] = (byte)(bytes[i] ^ '#');
    }

    public static void Reverse(UnmanagedMemoryAccessor data)
    {
        for(long i = 0, j = data.Capacity - 1; i < j; ++i, --j)
        {
            byte ib = data.ReadByte(i);
            byte jb = data.ReadByte(j);
            data.Write(i, jb);
            data.Write(j, ib);
        }
    }
}
